test/spec/IFrameManager-test.js

```diff
@@ -34,69 +34,60 @@ describe("iFrameManager", () => {
       const eventData = {
         origin: window.location.origin,
         data: {
-          type: messageType.SYNC
-        }
+          type: messageType.SYNC,
+        },
       };
 
       instance.onMessage(eventData);
       expect(instance.sendMessage).toHaveBeenCalledWith({
         type: messageType.SYNC_ACK,
         debugMode: undefined,
         version: "0.0.1",
-        iFrameUrl: IFrameManager.extractHostFromUrl(window.location.href)
+        iFrameUrl: IFrameManager.extractHostFromUrl(window.location.href),
       });
     });
   });
 
   describe("as iFrame host", () => {
-    function createHostInstanceAndTest(optionsOverrides, test) {
-      const options = Object.assign(
-        {
-          iFrameOptions: {
-            hostElementSelector: "body",
-            src: "http://a.b.c.d.e.f.c",
-            width: "100px",
-            height: "100px"
-          }
-        },
-        optionsOverrides
-      );
-      const instance = createInstance(options);
-      expect(instance).toBeDefined();
-
-      test(instance, options);
-
-      instance.remove();
-      const element = document.querySelector("iFrame");
-      expect(element).not.toBeTruthy();
-    }
-
     it("should validate iFrameOptions", () => {
       expect(() => new IFrameManager({ iFrameOptions: {} })).toThrow();
     });
 
     it("should get host element", () => {
       expect(IFrameManager.getHostElement("body")).toBe(
-        document.querySelector("body")
+        document.querySelector("body"),
       );
     });
 
-    it("should get origin", () => {
-      createHostInstanceAndTest({}, (instance, options) => {
-        expect(instance.destinationHost).toBe(options.iFrameOptions.src);
-      });
+    const options = {
+      iFrameOptions: {
+        hostElementSelector: "body",
+        src: "https://a.b.c.d.e.f.c",
+        width: "100px",
+        height: "100px",
+      },
+    };
+
+    it("should get origin",  done => {
+      spyOn(IFrameManager, "createIFrame").and.returnValue({ contentWindow: null });
```

Comment:
Documenting the reason for changing the tests:
We ran into an error using jasmine-browser-runner due to the async behavior of createIframe and iFrameManager.remove(). These changes let us test if the iFrame has been created with `options` without rendering it in the browser.
```
{
  "matcherName": "",
  "message": "TypeError: Cannot read properties of null (reading 'appendChild')",
  "stack": "    at HtmlReporter.jasmineDone (http://localhost:8888/__jasmine__/jasmine-html.js:203:13)\n    at <Jasmine>\n    at new Promise (<anonymous>)\n    at <Jasmine>",
  "passed": false,
  "expected": "",
  "actual": "",
  "globalErrorType": "afterAll"
}
```